src/model/reddit.ts

```diff
@@ -819,24 +817,35 @@ export async function getSearchArticle(data: IAPIData): Promise<IGenericResult<I
 export async function loginUser(data: IAPIData): Promise<any> {
     return new Promise<any>(async (resolve, reject) => {
         if (secureBaseHost && data.username && data.password) {
-            const rest: rm.RestClient = new rm.RestClient('login.user', secureBaseHost, undefined, { socketTimeout: data.timeout });
-            let url: string = "api/login/" + encodeURIComponent(data.username) + "?api_type=json&user=" + encodeURIComponent(data.username) + "&passwd=" + encodeURIComponent(data.password);
-            rest.create<any>(url, {
-                additionalHeaders: {
-                    "user-agent": "Reddit-Viewer / 2.0.0"
+            var request = require('request');
+            var options = {
+                'method': 'POST',
+                'url': 'https://ssl.reddit.com/api/login',
+                'headers': {
+                    'User-Agent': 'Reddit-Viewer / 2.0.0',
+                    'Content-Type': 'multipart/form-data'
+                },
+                formData: {
+                    'user': encodeURIComponent(data.username),
+                    'passwd': encodeURIComponent(data.password),
+                    'api_type': 'json'
                 }
-            })
-                .then(response => {
-                    console.log(response);
-                    if (response.statusCode === 200 && response.result && response.result.json) {
-                        resolve(response.result.json.data);
+            };
+            request(options, function (error: { message: any; }, response: { statusCode: string | number; result: { json: { data: any; }; }; body: any }) {
+                if (error) {
+                    throw reject(error.message);
+                }
+                if (response.statusCode === 200 && response.body) {
+                    var responseJson = JSON.parse( response.body ).json;
+                    if (responseJson.errors.length === 0) {
```

Comment:
This code should check for falsy responseJson before accessing errors property.
```suggestion
                    if (responseJson && responseJson.errors && responseJson.errors.length === 0) {
```
Or it is better to add a function to check for validity of response like isValidResponseJson
```suggestion
                    if (isValidResponseJson(responseJson)) {
```